frontend/src/cljs/imas_seamap/map/events.cljs

```diff
@@ -159,32 +150,38 @@
 (defn toggle-ignore-click [db _]
   (update-in db [:map :controls :ignore-click] not))
 
-(defn got-feature-info [db [_ request-id priority point info-format response]]
+(defn got-feature-info [db [_ request-id point info-format response]]
   (if (not= request-id (get-in db [:feature-query :request-id]))
-    db                           ; Ignore late responses to old clicks
-    (let [feature-info (case info-format
+    db ; Ignore late responses to old clicks
+    (let [db (update-in db [:feature-query :response-remain] dec)
+          {:keys [response-remain had-insecure? candidate]} (:feature-query db)
+
+          feature-info (case info-format
                          "text/html" (feature-info-html response)
                          "application/json" (feature-info-json response)
                          (feature-info-none))
-          db' (update-in db [:feature-query :response-remain] dec)
-          {:keys [response-priority response-remain candidate had-insecure?]} (:feature-query db')
-          higher-priority? (< priority response-priority)
-          candidate' (merge {:location point :had-insecure? had-insecure?} feature-info)]
-      (cond-> db'
-          ;; If this response has a higher priority, update the response candidate
-        higher-priority?
-        (assoc-in [:feature-query :candidate] candidate')
-          ;; If this is the last response expected, update the displayed feature
-        (zero? response-remain)
-        (assoc :feature (if higher-priority? candidate' candidate))))))
-
-(defn got-feature-info-error [db [_ _request_id priority _]]
-  (let [db (update-in db [:feature-query :response-remain] dec)
-        {:keys [response-priority response-remain]} (:feature-query db)
-        higher-priority? (< priority response-priority)]
-    (if (and (zero? response-remain) higher-priority?)
-      (assoc-in db [:feature :status] :feature-info/error)
-      db)))
+          feature-info (update feature-info :info str (:info candidate)) ;; Merge existing candidate info into retrieved info
```

Comment:
I think I'm all on-board apart from this line.  It _probably_ works, but just concatenating responses (assuming I'm not mis-reading, and it's just mashing strings together?) seems messy, and error-prone at some point (actually, what happens if you get a mix of html + json?).  I think better would be to collect a list of responses, and then we have more flexibility in rendering (it might be a big scrolling popup, it might be tabs with one response each, etc).

Reply:
I've removed the single "candidate" key from the query, and just store each server response in a list until all of them arrive, where they are then combined.
Doing this also allowed me to neaten up a lot of the code in got-feature-info and got-feature-info-error.
Happy for more feedback.